crates/storage/db/src/implementation/mdbx/tx.rs

```diff
@@ -234,13 +236,16 @@ impl<K: TransactionKind> DbTx for Tx<K> {
 
     fn commit(self) -> Result<bool, DatabaseError> {
         self.execute_with_close_transaction_metric(TransactionOutcome::Commit, |this| {
-            this.inner.commit().map_err(|e| DatabaseError::Commit(e.into()))
+            match this.inner.commit().map_err(|e| DatabaseError::Commit(e.into())) {
+                Ok((v, latency)) => (Ok(v), Some(latency)),
+                Err(e) => (Err(e), None),
+            }
```

Comment:
this looks a bit weird, feels like it would be better to put the latency into `Ok` variant and make it non-optional. But then we'd need to replace the `Ok` variant in `execute_with_close_transaction_metric`. What do you think is better?

Reply:
@shekhirin i'm not sure if it is ok to make it non-optional, since `abort` function does not provide any latencies. 
We can however introduce a method to `CommitLatency` that requires self-invocation (thus, it must be called after passing an inner pointer to some FFI method). This change would make getter methods return `Some(Duration)` instead of `None` (so the getters' return type must also be altered). However, this could lead to less readable code due to the need for additional checks within each metric during record. To mitigate this, we could employ a useful macro.

Additionally, we could employ the proposed method to set an internal flag, checked during transaction metric recording. Yet, I'm concerned this might not be a best practice.

Each of these approaches would enable the initialization of a new `CommitLatency` struct within the `abort` function, allowing us to pass it as part of the `Ok` variant